src/custom/components/Popups/PopupItem.tsx

```diff
@@ -0,0 +1,106 @@
+import React, { useCallback, useContext, useEffect } from 'react'
+import { X } from 'react-feather'
+import { useSpring } from 'react-spring/web'
+import styled, { ThemeContext } from 'styled-components'
+import { animated } from 'react-spring'
+import { PopupContent } from 'state/application/actions'
+import { useRemovePopup } from 'state/application/hooks'
+import ListUpdatePopup from 'components/Popups/ListUpdatePopup'
+import TransactionPopup from 'components/Popups/TransactionPopup'
+import MetaTransactionPopup from 'components/Popups/MetaTransactionPopup'
+
+export const StyledClose = styled(X)`
+  position: absolute;
+  right: 10px;
+  top: 10px;
+
+  :hover {
+    cursor: pointer;
+  }
+`
+export const Popup = styled.div`
+  display: inline-block;
+  width: 100%;
+  padding: 1em;
+  background-color: ${({ theme }) => theme.bg1};
+  position: relative;
+  border-radius: 10px;
+  padding: 20px;
+  padding-right: 35px;
+  overflow: hidden;
+
+  ${({ theme }) => theme.mediaWidth.upToSmall`
+    min-width: 290px;
+    &:not(:last-of-type) {
+      margin-right: 20px;
+    }
+  `}
+`
+const Fader = styled.div`
+  position: absolute;
+  bottom: 0px;
+  left: 0px;
+  width: 100%;
+  height: 2px;
+  background-color: ${({ theme }) => theme.bg3};
+`
+
+const AnimatedFader = animated(Fader)
+
+export default function PopupItem({
+  removeAfterMs,
+  content,
+  popKey
+}: {
+  removeAfterMs: number | null
+  content: PopupContent
+  popKey: string
+}) {
+  const removePopup = useRemovePopup()
+  const removeThisPopup = useCallback(() => removePopup(popKey), [popKey, removePopup])
+  useEffect(() => {
+    if (removeAfterMs === null) return undefined
+
+    const timeout = setTimeout(() => {
+      removeThisPopup()
+    }, removeAfterMs)
+
+    return () => {
+      clearTimeout(timeout)
+    }
+  }, [removeAfterMs, removeThisPopup])
+
+  const theme = useContext(ThemeContext)
+
+  let popupContent
+  if ('txn' in content) {
+    const {
+      txn: { hash, success, summary }
+    } = content
+    popupContent = <TransactionPopup hash={hash} success={success} summary={summary} />
+  } else if ('listUpdate' in content) {
+    const {
+      listUpdate: { listUrl, oldList, newList, auto }
+    } = content
+    popupContent = <ListUpdatePopup popKey={popKey} listUrl={listUrl} oldList={oldList} newList={newList} auto={auto} />
+  } else if ('metatxn' in content) {
+    const {
+      metatxn: { id, success, summary }
+    } = content
+    popupContent = <MetaTransactionPopup id={id} success={success} summary={summary} />
+  }
```

Comment:
This is the only part I changed, everything else is copy-:spaghetti:

Reply:
probably best to create a `PopupItemMod.tsx` and import the styled components

Reply:
maybe, but as it is a copypasta component with just a couple lines changed, let's not